Library/Homebrew/cmd/outdated.rb

```diff
@@ -37,4 +33,30 @@ def outdated_brews(formulae)
       end
     end.compact
   end
+
+  def formulae_to_check
+    ARGV.resolved_formulae.any? ? ARGV.resolved_formulae : Formula.installed
+  end
+
+  def print_outdated
+    outdated_brews(formulae_to_check) do |f, versions|
+      if ($stdout.tty? || ARGV.verbose?) && !ARGV.flag?("--quiet")
+        puts "#{f.full_name} (#{versions*', '} < #{f.pkg_version})"
+      else
+        puts f.full_name
+      end
+    end
+  end
+
+  def print_outdated_json
+    json = []
+    outdated = outdated_brews(formulae_to_check) do |f, versions|
+      json << {name: f.name,
+               versions: versions.collect(&:to_s),
+               pkg_version: f.pkg_version.to_s}
+    end
+    puts Utils::JSON.dump(json)
+    
+    outdated
```

Comment:
It's a side effect I don't really like, but was effectively added between when I made my change last year and when I rebased it now. The check on line 11 didn't really make sense to duplicate, so I made both `print_outdated_json` and `print_outdated` return the output of `outdated_brews` in the same manner as line 8 of the previous revision. I'm explicitly returning outdated here because `puts` returns `nil`.